src/dos/program_more.cpp

```diff
@@ -0,0 +1,335 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "program_more.h"
+
+#include "../ints/int10.h"
+#include "callback.h"
+#include "checks.h"
+#include "string_utils.h"
+
+#include <algorithm>
+#include <cctype>
+#include <vector>
+
+CHECK_NARROWING();
+
+
+// ASCII control characters
+static constexpr char code_ctrl_c    = 0x03; // end of text
+static constexpr char code_lf        = 0x0a; // line feed
+static constexpr char code_cr        = 0x0d; // carriage return
+static constexpr char code_esc       = 0x1b; // escape
+
+
+void MORE::Run()
+{
+	// Handle command line
+	if (HelpRequested()) {
+		WriteOut(MSG_Get("SHELL_CMD_MORE_HELP_LONG"));
+		return;
+	}
+	if (!ParseCommandLine())
+		return;
+
+	// Check screen height
+	constexpr uint16_t min = 10;
+	new_line_counter_start = std::max(min, INT10_GetTextRows());
+	new_line_counter_start = static_cast<uint16_t>(new_line_counter_start - 2);
```

Comment:
Looks like more than a check, because we're saving the result as some minimum starting position? 
The -2 strips off some extra character(s) (or something?).

(Maybe add a comment about these little details).

Reply:
@kcgen This is still a draft, not for review - for example, reading files is not implemented yet. And this part will actually change, as I have some idea how to make it a little more resistant to ANSI control sequences.

Reply:
Added a comment explaining why.